File: packages/lib-frontend-js/src/route/utils/trimPathname/trimPathname.ts
import {
  type TrimPathnameModel,
  type TrimPathnameParamsModel,
} from '@lib/frontend/route/utils/trimPathname/trimPathname.models';
import { slug } from '@lib/shared/core/utils/slug/slug';
import trim from 'lodash/trim';

export const trimPathname = (
  ...[value, options = {}]: TrimPathnameParamsModel
): TrimPathnameModel => {
  if (value === '*') return value;
  const isPrefix = options?.isPrefix ?? true;
  const isSlug = options?.isSlug ?? true;
  const pathname = value
    .split('/')
    .filter(Boolean)
    .map((char) => {
      let v = trim(char, '/');
      isSlug && (v = v.replace(/\w\S*/g, slug));
      return v;
    })
    .join('/');
  const result = trim(pathname, '/');
  return isPrefix ? `/${result}` : result;
};

File: packages/lib-shared-py/src/lib_shared/route/utils/trim_pathname.py
from pydash import kebab_case, trim_start

from lib_shared.route.utils.trim_pathname_models import (
    TrimPathnameModel,
    TrimPathnameParamsModel,
)


def trim_pathname(params: TrimPathnameParamsModel) -> TrimPathnameModel:
    pathname = trim_start(params, "/")
    pathname = map(kebab_case, pathname.split("/"))
    pathname = "/".join(pathname)
    return f"/{pathname}"
